### PR TITLE
Fix iOS navbar title color not resetting correctly on pop

### PR DESCRIPTION
**Pull request by mpiannucci**
Brute force, but it seems to work really well. I tried some other solutions from online but this is the only one that seemed to work. I have never seen this bug before any where else.

### CODE REVIEW

File: NavigationReactNative/src/ios/NVSceneController.m [hunk 1]
```diff
@@ -47,14 +46,23 @@ - (void)viewWillAppear:(BOOL)animated
...
         [self.navigationItem setLargeTitleDisplayMode:navigationBar.largeTitle ? UINavigationItemLargeTitleDisplayModeAlways : UINavigationItemLargeTitleDisplayModeNever];
     }
 }
 
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    
+    // This is seemingly required, otherwise the small navbar title color does not get reset when only called from viewWillAppear
+    NVNavigationBarView *navigationBar = (NVNavigationBarView *) [self.view viewWithTag:NAVIGATION_BAR];
+    [navigationBar updateTitleColors];
```
Comment (grahammendick) [inline NavigationReactNative/src/ios/NVSceneController.m]: Do we need a separate `updateTitleColors`? Why not call `updateColors` again? I know that we don't need to update the tint colors but it doesn't cost anything

File: NavigationReactNative/src/ios/NVSceneController.m [hunk 2]
```diff
@@ -47,14 +46,23 @@ - (void)viewWillAppear:(BOOL)animated
...
     if (@available(iOS 11.0, *)) {
         self.navigationController.navigationBar.prefersLargeTitles = true;
         [self.navigationItem setLargeTitleDisplayMode:navigationBar.largeTitle ? UINavigationItemLargeTitleDisplayModeAlways : UINavigationItemLargeTitleDisplayModeNever];
     }
 }
 
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    
+    // This is seemingly required, otherwise the small navbar title color does not get reset when only called from viewWillAppear
```
Comment (grahammendick) [inline NavigationReactNative/src/ios/NVSceneController.m]: Thanks but I'm not a fan of code comments. This is great detail though so you can add it to the PR's description

File: NavigationReactNative/src/ios/NVSceneController.m [hunk 3]
```diff
@@ -1,15 +1,14 @@
 #import "NVSceneController.h"
-#import "NVSceneView.h"
 #import "NVNavigationBarView.h"
 #import "NVSearchBarView.h"
 
 @implementation NVSceneController
 {
-    NVSceneView *_view;
+    UIView *_view;
```
Comment (grahammendick) [inline NavigationReactNative/src/ios/NVSceneController.m]: You're working off an old version of the `NVSceneController`. Make sure you pull from master

File: NavigationReactNative/src/ios/NVSceneController.m (hunk 1 above)
Comment (mpiannucci) [reply NavigationReactNative/src/ios/NVSceneController.m]: Yeah it doesn't hurt anything, was just trying to cut down on duplicate calls

**Comment by mpiannucci** [comment]
Synced up with master and just made the two little changes there. Hope thats better!

**Comment by grahammendick** [comment]
Looks great, thanks. In case you missed it, I released v5.5.0 with your Navigation Bar and Tab Bar color changes in!

**Comment by mpiannucci** [comment]
Yes I saw!!! thank you so much!!!!

**Pull request merged**